core/src/main/scala/org/apache/spark/deploy/client/AppClient.scala

```diff
@@ -77,6 +77,16 @@ private[spark] class AppClient(
     private val registrationRetryThread =
       ThreadUtils.newDaemonSingleThreadScheduledExecutor("appclient-registration-retry-thread")
 
+    private val receiveAndReplyMaxPoolSize = conf.getInt(
```

Comment:
We should consider adding a JIRA to follow up and document this new param. Also might be good to have a comment explaining why 3 is the default (at first read through I'm not sure why it is).